source/common/http/utility.h

```diff
@@ -157,16 +172,15 @@ Http1Settings parseHttp1Settings(const envoy::api::v2::core::Http1ProtocolOption
  * @param body_text supplies the optional body text which is sent using the text/plain content
  *                  type.
  * @param grpc_status the gRPC status code to override the httpToGrpcStatus mapping with.
- * @param is_head_request tells if this is a response to a HEAD request
  */
-void sendLocalReply(bool is_grpc, StreamDecoderFilterCallbacks& callbacks, const bool& is_reset,
-                    Code response_code, absl::string_view body_text,
-                    const absl::optional<Grpc::Status::GrpcStatus> grpc_status,
-                    bool is_head_request);
+void sendLocalReply(const LocalReplyInfo& info, StreamDecoderFilterCallbacks& callbacks,
```

Comment:
This might be a good opportunity to improve this API a bit... For example, the optional `grpc_status` parameter isn't relevant a lot of the time. And sometimes you might want your local reply to have more structure than just a `body_text` string. What do you think?

Reply:
Yes, and I hope we can be multiple iterations to improve this API, rather than one time.